tools/python/local_ads/mwm_to_csv_4localads.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python2.7
+import os
+import sys
+
+# TODO: Make mwm an installable module.
+sys.path.append(
+    os.path.join(
+        os.path.dirname(__file__), '..', 'mwm'
+    )
+)
+
+import argparse
+import csv
+import mwm
+import logging
+import ctypes
+from zlib import adler32
+from multiprocessing import Pool, Queue, Process
+
+
+HEADERS = {
+    'mapping': 'osmid fid mwm_id mwm_version'.split(),
+    'sponsored': 'stype sid fid mwm_id mwm_version'.split()
+}
+QUEUES = {name: Queue() for name in HEADERS}
+GOOD_TYPES = ("amenity", "shop", "tourism", "leisure", "sport",
+              "craft", "man_made", "office", "historic")
+
+
+def generate_id_from_name_and_version(name, version):
+    return ctypes.c_long((adler32(name) << 32) | version).value
+
+
+def parse_mwm(mwm_name, osm2ft_name, oversion, types_name):
+    with open(osm2ft_name, 'rb') as f:
+        ft2osm = mwm.read_osm2ft(f, ft2osm=True, tuples=False)
+    with open(mwm_name, 'rb') as f:
+        reg = mwm.MWM(f)
+        version = oversion or reg.read_version()['version']
+        mwm_id = generate_id_from_name_and_version(mwm_name, version)
+        reg.read_header()
+        reg.read_types(types_name)
+        for feature in reg.iter_features(metadata=True):
+            osm_id = ft2osm.get(feature['id'], None)
+            if osm_id is None:
+                if 'ref:sponsored' in feature['metadata']:
+                    for t in feature['header']['types']:
+                        if t.startswith('sponsored-'):
+                            QUEUES['sponsored'].put((t[t.find('-')+1:], feature['metadata']['ref:sponsored'], feature['id'], mwm_id, version))
+                            break
+            else:
+                for t in feature['header']['types']:
+                    if t.startswith(GOOD_TYPES):
+                        QUEUES['mapping'].put((ctypes.c_long(osm_id).value, feature['id'], mwm_id, version))
+                        break
+
+
+def write_csv(output_dir, qtype):
+    with open(os.path.join(output_dir, qtype + '.csv'), 'w') as f:
+        mapping = QUEUES[qtype].get()
+        w = csv.writer(f)
+        w.writerow(HEADERS[qtype])
```

Comment:
Adding header requires changes in loading command. One should add HEADER to COPY command for psql.